rust/cardano-chain-follower/src/mithril_snapshot_iterator.rs

```diff
@@ -29,6 +29,8 @@ struct MithrilSnapshotIteratorInner {
     start: Point,
     /// Previous iteration point.
     previous: Point,
+    /// Mithrill snapshot directory path
+    path: PathBuf,
```

Comment:
Move this to the `MithrilSnapshotIterator`  it's not mutable data, so we don't need to Mutex it inside the Arc.